src/backend/distributed/utils/jsonbutils.c

```diff
@@ -56,6 +56,11 @@ ExtractFieldJsonb(Datum jsonbDoc, const char *fieldName, Datum *result, bool as_
 	InitFunctionCallInfoData(*functionCallInfo, &fmgrInfo, 2, DEFAULT_COLLATION_OID, NULL,
 							 NULL);
 
+	if (jsonbDoc == 0)
```

Comment:
Given that DistNodeMetadata() asserts that returned datum is not null, this doesn't seem to be right way of fixing the memory issue, will take a closer look to see what's happening here.

Reply:
> we need to copy jsonb datum out before closing the table.

Indeed I'm almost sure that this should be the reason. We weren't hitting to this memory error before probably because statistics collection is not enabled for valgrind tests (needs to be verified), where statistics collector was the only caller of DistNodeMetadata() before the you changes made for citus_add_node().